src/VirtoCommerce.CatalogModule.Data/Services/AssociationService.cs

```diff
@@ -78,8 +92,57 @@ public async Task SaveChangesAsync(IHasAssociations[] owners)
                 source.Associations.Patch(target.Associations, associationComparer, (sourceAssociation, targetAssociation) => sourceAssociation.Patch(targetAssociation));
 
                 await repository.UnitOfWork.CommitAsync();
+                //Reset cached associations
+                ItemCacheRegion.ExpireRegion();
```

Comment:
Could you add the single ClearCache(string productIds) method which clears the cache for the array of product ids. Also it would be good to add `ItemCacheRegion.ExpireProduct(string productId)` method in ItemCacheRegion

Reply:
To prevent the whole cache region clearing we need to find what objects are `CatalogProduct`s and get their ids.